packages/request-client.js/src/api/request.ts

```diff
@@ -78,6 +80,21 @@ export default class Request {
    */
   private currencyManager: CurrencyTypes.ICurrencyManager;
 
+  /**
+   * Transaction data of a in-memory request, necesary for persisting the request later on.
+   */
+  public readonly transactionData: DataAccessTypes.ITransaction | undefined;
+
+  /**
+   * Topics of a in-memory request, necesary for persisting the request later on.
```

Comment:
```suggestion
   * Topics of an in-memory request, necessary for persisting the request later on.
```

Since you have 3 attributes with the same presence constraint (should be defined for in-memory, or undefined), wouldn't it help merging them in 1 "inMemoryInfo" object, that we can validate once, and the 3 attributes are required within it?

Reply:
If the suggestion above is not justified, please mind the attributes and constructor arguments names: although they are super clear in the context of this PR, with less context it looks odd to pass a `paymentRequest` to a request constructor (for example).

Reply:
I agree that merging them would be a better experience.

Reply:
Done :white_check_mark: